pkg/controller/common/reconciler/results_test.go

```diff
@@ -216,3 +217,44 @@ func TestResults_IsReconciled(t *testing.T) {
 		})
 	}
 }
+
+func TestResults_WithRequeue(t *testing.T) {
+	type args struct {
+		requeueAfter []time.Duration
+	}
+	tests := []struct {
+		name string
+		args args
+		want reconcile.Result
+	}{
+		{
+			name: "default requeue",
+			args: args{
+				requeueAfter: nil,
+			},
+			want: reconcile.Result{Requeue: false, RequeueAfter: 10 * time.Second},
+		},
+		{
+			name: "no requeue",
+			args: args{
+				requeueAfter: []time.Duration{0},
+			},
+			want: reconcile.Result{Requeue: false, RequeueAfter: 0},
+		},
```

Comment:
So technically you can do a non-requeue result with this function. I wonder if we should disallow it though as it seems confusing.

Reply:
Yes, it's not great. I'm not sure we should return an error in that case though, maybe adding a comment to explain that `0` means "no requeue at all" in the godoc is enough (we could also log an error to surface when it happens?).

Reply:
What we could do is enforce a requeue to the default requeue after value if the passed value is 0. This would also handle the bug you mentioned in the context of certificate expiration

Reply:
> This would also handle the bug you mentioned in the context of certificate expiration

Actually not quite as that uses the `ReconciliationState` struct to manage the requeue. But I chose to fix the issue you found as well given that is roughly fits into the realm of requeue refactoring that we are doing in this PR. Lmk if you have concerns.

Reply:
LGTM 👍